src/UnitfulRecipes.jl

```diff
@@ -12,6 +12,11 @@ function recipe!(attr, arr)
 end
 
 function recipe!(attr, arrs...)
+    if haskey(attr, :marker_z)
+        u = unit(eltype(attr[:marker_z]))
+        attr[:marker_z] = ustrip.(u, attr[:marker_z])
+        attr[:colorbar_title] = string(u)
```

Comment:
What does this do if `marker_z` is unitless?

Reply:
I guess it leaves it as is? Again sorry this is dirty...

Reply:
It's a honest question, I guess `u == NoUnits()` and becomes an empty string?

Reply:
Yes I just checked and that's what happens. (sorry for the delay I was sleeping — Australian time here.)